drivers/bluetooth/hci/spi.c

```diff
@@ -411,37 +504,31 @@ static int bt_spi_send(struct net_buf *buf)
 		return -EINVAL;
 	}
 
-	/* Poll sanity values until device has woken-up */
-	do {
-		kick_cs();
-		ret = bt_spi_transceive(header_tx, 5, header_rx, 5);
+	ret = bt_spi_get_header(SPI_WRITE, &size);
+	size = MIN(buf->len, size);
 
-		/*
-		 * RX Header must contain a sanity check Byte and size
-		 * information.  If it does not contain BOTH then it is
-		 * sleeping or still in the initialisation stage (waking-up).
-		 */
-	} while ((header_rx[STATUS_HEADER_READY] != READY_NOW ||
-		  (header_rx[1] | header_rx[2] | header_rx[3] | header_rx[4]) == 0U) && !ret);
+	if (size < buf->len) {
+		LOG_WRN("Unable to write full data");
```

Comment:
Writing partial data is worse than writing no data at all IMO.
This check can also never fail for the non-ST protocol, not sure whether the compiler is smart enough to figure that out though.

Reply:
Done.